app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java

```diff
@@ -0,0 +1,203 @@
+        intent.putExtra("receiver", mResultReceiver);
+
+        startService(intent);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_route_detail, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                this.finish();
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onReceiveResult(int resultCode, Bundle resultData) {
+        switch (resultCode) {
+            case DownloadIntentService.STATUS_FINISHED:
+                mRouteNameTextView.setText(mRouteId + " - " + mRouteName);
+                String resultBus = resultData.getString(DownloadIntentService.RESULT_BUS);
+                String resultDepart = resultData.getString(DownloadIntentService.RESULT_DEPARTURE);
+
+                try {
+                    // Set BusStop Adapter
+                    mBusStopList = new BusStopList(resultBus);
+                    BusStopAdapter busAdapter = new BusStopAdapter(mBusStopList.getBusStopList());
+                    mStopListView.setAdapter(busAdapter);
+
+                    // List of Departures
+                    mDepartureList = new DepartureList(resultDepart);
+
+                    // set Weekday Departure Adapter
+                    ArrayList<Departure> weekdayDeparts = mDepartureList.getDepartures("WEEKDAY");
+                    if (weekdayDeparts.isEmpty()) {
+                        weekdayDeparts.add(new Departure(0, "WEEKDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter weekdayAdapter = new DepartureAdapter(weekdayDeparts);
+                    mWeekdayListView.setAdapter(weekdayAdapter);
+
+                    // set Saturday Departure Adapter
+                    ArrayList<Departure> saturdayDeparts = mDepartureList.getDepartures("SATURDAY");
+                    if (saturdayDeparts.isEmpty()) {
+                        saturdayDeparts.add(new Departure(0, "SATURDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter saturdayAdapter = new DepartureAdapter(saturdayDeparts);
+                    mStaturdayListView.setAdapter(saturdayAdapter);
+
+                    // set Sunday Departure Adapter
+                    ArrayList<Departure> sundayDeparts = mDepartureList.getDepartures("SUNDAY");
+                    if (sundayDeparts.isEmpty()) {
+                        sundayDeparts.add(new Departure(0, "SUNDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter sundayAdapter = new DepartureAdapter(sundayDeparts);
+                    mSundayListView.setAdapter(sundayAdapter);
+                } catch (JSONException e) {
+                    e.printStackTrace();
```

Comment:
Could create a NoDepartureException class that extend Exception class ant
trow it when the list of departures is empty.

2015-06-17 13:13 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In
> app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java
> <https://github.com/daniboy000/at_challenge/pull/1#discussion_r32644183>:
>
> > +                    ArrayList<Departure> saturdayDeparts = mDepartureList.getDepartures("SATURDAY");
> > +                    if (saturdayDeparts.isEmpty()) {
> > +                        saturdayDeparts.add(new Departure(0, "SATURDAY", "NO DEPARTURES"));
> > +                    }
> > +                    DepartureAdapter saturdayAdapter = new DepartureAdapter(saturdayDeparts);
> > +                    mStaturdayListView.setAdapter(saturdayAdapter);
> > +
> > +                    // set Sunday Departure Adapter
> > +                    ArrayList<Departure> sundayDeparts = mDepartureList.getDepartures("SUNDAY");
> > +                    if (sundayDeparts.isEmpty()) {
> > +                        sundayDeparts.add(new Departure(0, "SUNDAY", "NO DEPARTURES"));
> > +                    }
> > +                    DepartureAdapter sundayAdapter = new DepartureAdapter(sundayDeparts);
> > +                    mSundayListView.setAdapter(sundayAdapter);
> > +                } catch (JSONException e) {
> > +                    e.printStackTrace();
>
> It's not good to swallow exceptions like this. Couldn't you handle that in
> the BusStopList class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/daniboy000/at_challenge/pull/1/files#r32644183>.
>



-- 
Daniel Besen de Aguiar

"Those who wish to succeed must ask the right preliminary questions"


      Aristotle